Swift/Database.swift

```diff
@@ -165,16 +152,20 @@ public final class Database {
         _ document: MutableDocument, conflictHandler: @escaping (MutableDocument, Document?) -> Bool
     ) throws -> Bool {
         do {
-            return try defaultCollectionOrThrowEx().save(document: document,
-                                                         conflictHandler: conflictHandler)
+            try _impl.save(
+                document._impl as! CBLMutableDocument,
+                conflictHandler: { (cur: CBLMutableDocument, old: CBLDocument?) -> Bool in
+                    let col = try? self.defaultCollection()
```

Comment:
Same as above, need to throw NotOpen error when default collection is nil.

Reply:
will throw exception